.github/workflows/checks.yml

```diff
@@ -8,6 +8,10 @@ on:
   schedule:
     - cron: '0 23 * * SUN-THU'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
```

Comment:
If [this comment](https://github.com/optuna/optuna/pull/3095#discussion_r756781164) is true, I don't think the concurrency rule is necessary because this workflow has only a single job.

Reply:
Here, we want to limit the number of running jobs for each worin the PR as `1`. 

For example, in this PR case, the `group` would be something like `group: {{checks}}-{{refs/heads/feature/set-concurrency-to-ga}}`. And if we have two workflows which have the same name of group, the former will be killed.

The situation will happen if we push two commits on this PR in a short term. And, the check test for the first commit will be killed.

Reply:
> The situation will happen if we push two commits on this PR in a short term.

Ah, that makes sense. Then, we must include this concurrency setting here. Thank you for your explanation.